numba/targets/codegen.py

```diff
@@ -120,8 +120,10 @@ def _optimize_final_module(self):
         """
         Internal: optimize this library's final module.
         """
+
         self._codegen._mpm.run(self._final_module)
         self._final_module = remove_redundant_nrt_refct(self._final_module)
+        self._codegen._mpm.run(self._final_module)
```

Comment:
I am uncomfortable with this extra optimization pass run without more benchmark data.  There have been complaints that compile time is slow for some folks.  The extra opt-pass would be too expensive for functions that are already taking minutes to compile.